### PR TITLE
Gh 870 conditional none handling

### PR DESCRIPTION
**Pull request by j8934893**

### CODE REVIEW

File: python-shell/src/gafferpy/gaffer_operations.py [hunk 1]
```diff
@@ -2623,18 +2623,16 @@ class Conditional(ToJson, ToCodeString):
 
     def __init__(self, predicate=None, transform=None):
 
-        if predicate is not None:
-            if not isinstance(predicate, gaffer_predicates.Predicate):
+        if predicate is not None and not isinstance(predicate, gaffer_predicates.Predicate):
                 self.predicate = JsonConverter.from_json(predicate,
```
Comment (t92549) [inline python-shell/src/gafferpy/gaffer_operations.py:2627]: Indentation

File: python-shell/src/gafferpy/gaffer_operations.py [hunk 2]
```diff
@@ -2623,18 +2623,16 @@ class Conditional(ToJson, ToCodeString):
...
                                                          gaffer_predicates.Predicate)
-            else:
-                self.predicate = predicate
+        else:
+            self.predicate = predicate
 
-        if transform is not None:
-            if not isinstance(transform, Operation):
+        if transform is not None and not isinstance(transform, Operation):
                 self.transform = JsonConverter.from_json(transform, Operation)
```
Comment (t92549) [inline python-shell/src/gafferpy/gaffer_operations.py:2633]: Indentation

**Review by t92549 — CHANGES_REQUESTED** [review]

**Comment by t92549** [comment]
Happy to merge after small indentation fixes. Closes #870

**Comment by n3101** [comment]
@j8934893 Hiya, we're looking to do a release in a couple of weeks. If you make the requested changes here soonish, we can include this item.

**Comment by j8934893** [comment]
Git appears to have done something odd with my local branch so split this into a new PR - https://github.com/gchq/gaffer-tools/pull/897

Sorry for being a pain.

**Comment by t92549** [comment]
Closing as superseded by #898

**Pull request closed without merging**